### PR TITLE
Document inrepoconfig

### PR DESCRIPTION
**Pull request by alvaroaleman**
This PR adds documentation for the inrepoconfig feature.

### CODE REVIEW

File: prow/inrepconfig.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+```
+in_repo_config:
+  enabled:
+    # The key can be one of "*" for "globally", "org" for the GitHub organization or "org/repo" for
+    # a specific repo. The narrowest match is used.
+    "kubernetes/kubernetes": true
+```
+
+Afterwards, you need to add a config verification job to make sure people people get told about
```
Comment (stevekuznetsov) [inline prow/inrepconfig.md]: Do we want to ship like this? Why not just dynamically generate this presubmit?

File: prow/inrepconfig.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      base_ref: master
+      # Omit when using a publicly accessible repo
+      clone_uri: git@github.com:kubernetes/test-infra.git
+     spec:
+       containers:
+       - image: gcr.io/k8s-prow/checkconfig:v20191205-050b151d0
+         command:
+         - /app/prow/cmd/checkconfig/app.binary
+         args:
+         - --plugin-config=/home/prow/go/src/github.com/kubernetes/test-infra/path/to/plugins.yaml
```
Comment (stevekuznetsov) [inline prow/inrepconfig.md]: nit: your working directory is the root of your main clone in a GOPATH-style env so you can make this clearer by using a relative path

File: prow/inrepconfig.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+         - --config-path=/home/prow/go/src/github.com/kubernetes/test-infra/path/to/config.yaml
+         - --prow-yaml-repo-name=$(REPO_OWNER)/$(REPO_NAME
```
Comment (stevekuznetsov) [inline prow/inrepconfig.md]: nit: this is broken

File: prow/inrepconfig.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+job in the central repository rather than the code repository, so the `checkconfig` version used
+stays in sync with the Prow version used. It looks like this:
+
+```
+presubmits:
+  kubernetes/kubernetes:
+  - name: pull-kubernetes-validate-prow-yaml
+    always_run: true
+    decorate: true
+    # Omit when using a publicly accessible repo
```
Comment (stevekuznetsov) [inline prow/inrepconfig.md]: This is the common case so maybe don't show it in the example?

Comment (stevekuznetsov) [reply prow/inrepconfig.md]: And also we have OAuth support landing so you might not need it anyway

File: prow/inrepconfig.md (hunk 1 above)
Comment (alvaroaleman) [reply prow/inrepconfig.md]: Well, primarily this documents the current state of affairs. This doesn't mean we can not improve it, but thats how it is today.

**Comment by alvaroaleman** [comment]
@stevekuznetsov comments addressed, PTAL

**Pull request merged**